git_trac/releasemgr/app.py

```diff
@@ -185,15 +191,39 @@ def test_merge(self, ticket_number):
                 raise ValueError('merge was not clean')
             self._commit(commit_message)
 
-    def merge_multiple(self, ticket_numbers, **kwds):
-        for ticket_number in ticket_numbers:
-            self.merge(ticket_number, **kwds)
+    def merge_multiple(self, ticket_numbers, limit=0, **kwds):
+        successful = []
+        errors = []
+        try:
+            for ticket_number in ticket_numbers:
+                try:
+                    self.merge(ticket_number)
+                    successful.append(ticket_number)
+                except ValueError as err:
+                    errors.append((ticket_number, str(err)))
+                    if u'ticket dependencies' not in str(err) and u'already merged' not in str(err):
+                        comment = 'Merge failure on top of:\n\n{}\n\n{}'.format(
+                            self.describe().replace('\n', '\n\n'), err)
+                        self.set_ticket_to_needs_work(ticket_number, comment)
+                if limit and (len(successful) >= limit):
+                    break
+        finally:
+            print("\n### Summary ###\n")
+            if len(ticket_numbers) != 1:
```

Comment:
Why not print the result if there is only a single ticket attempted? Especially if there was an error I'll want to know.

Reply:
OK